library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableStateStore.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.state.spillable;
+
+import org.apache.apex.malhar.lib.state.BucketedState;
+
+import com.datatorrent.api.Component;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Operator;
+
+public interface SpillableStateStore extends BucketedState, Component<Context.OperatorContext>,
```

Comment:
Maybe I am missing something but please bear with my questions:
1. Why should a WindowedOperator not have any knowledge of SpillableComponents?
2. Let's say if above is the case, then what is going to be the API of WindowedOperator with respect to storage. If WindowedOperator is specifying the contract for a pluggable storage than an implementation of SpillableStateStore can implement that and therefore can be set on WindowedOperator. However,  I think SpillableStateStore  itself can be that contract.